versioned/tiered/tiered-impl/src/main/java/org/projectnessie/versioned/impl/TieredVersionStore.java

```diff
@@ -111,8 +122,20 @@ public TieredVersionStore(StoreWorker<DATA, METADATA, DATA_TYPE> storeWorker, St
     // We actually do not need an executor at all, when waitOnCollapse==true. Unit tests use
     // waitOnCollapse==true. It is not nice to instantiate an executor-service but never shut it
     // down, like unit tests did.
-    this.executor =  waitOnCollapse ? MoreExecutors.directExecutor() : Executors.newCachedThreadPool();
+    Executor executor;
+    if (waitOnCollapse) {
+      executor = MoreExecutors.directExecutor();
+    } else {
+      executor = Executors.newCachedThreadPool();
+      executor = ExecutorServiceMetrics.monitor(Metrics.globalRegistry, executor, "TieredVersionStore");
+    }
+    this.executor = executor;
     this.waitOnCollapse = waitOnCollapse;
+
+    this.gauges = ImmutableMap.<String, Supplier<Number>>builder()
```

Comment:
similar to above, can't we just get the metrics registry and register these rather than pushing to the interface. As an aside, I don't understand why we reset the counter when this gauge supplier is called.